sotrplib/handlers/base.py

```diff
@@ -207,6 +210,12 @@ def _run(self) -> tuple[list[list], list[object], list[ProcessableMap]]:
         decorated with the flow as prefect needs these to be defined in advance.
         """
         self.maps = self.basic_task(self.build_map).map(self.maps).result()
+        map_freqs = list(set([m.frequency for m in self.maps]))
+        self.maps += [
+            c.coadd(self.maps, frequency=freq)
+            for c in self.coadders
+            for freq in map_freqs
+        ]
```

Comment:
It seems that this should probably be done within the tasking framework rather than on the main thread... @ColmTalbot ?